sgx-sdk, sgx-psw: 2.14 -> 2.15.1

veehaitch opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Also add some of the new samples as tests. Disable parallel builds for
the samples as they don't seem to support it (fail randomly).

~**Still a draft. Waiting for #148593.**~

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

blitz commented on pkgs/os-specific/linux/sgx-sdk/samples.nix:
```diff
@@ -12,7 +12,7 @@ let
     buildInputs = [
       sgx-sdk
     ];
-    enableParallelBuilding = true;
```
nit: Was parallel building buggy?

veehaitch replied:
Yes, it failed randomly. Also, it's not too important to have it for the samples: they're tiny and build quickly.

Mic92 replied:
Please switch to `false` instead and maybe add a comment. There are efforts to build all packages eventually in parallel.

veehaitch replied:
That makes sense, thanks. Force-pushed [a new version](https://github.com/NixOS/nixpkgs/compare/124ad4ba0539f59ebaef0f1095912e8d928eb65a..4f7f8d0b2dce31038d1fde0d5d8321e68f16e0c6).

veehaitch committed:
sgx-sdk, sgx-psw: 2.14 -> 2.15.1

Also add some of the new samples as tests. Disable parallel builds for
the samples as they don't seem to support it (fail randomly).

veehaitch commented:
Rebased, as https://github.com/NixOS/nixpkgs/pull/148593 is merged. I didn't test the AESM service on an SGX machine yet.

veehaitch commented:
> I didn't test the AESM service on an SGX machine yet.

Works fine!

blitz commented:
@veehaitch The AESM daemon runs fine. `sgx-detect` is happy after manually linking `/dev/sgx_enclave` to `/dev/sgx/enclave`. Weird. Not sure why the device node changed, but this is unrelated to the SGX userspace code.

Good to merge from my end.
